src/components/message.vue

```diff
@@ -0,0 +1,588 @@
+          setIsEdit(false),
+            changeCommentText({
+              id: comment.id,
+              newCommentText: textareaValue,
+            })
+        "
+      >
+        UPDATE
+      </button>
+      <div class="modal_container">
+        <div class="overlay" v-if="showModal">
+          <div class="modal">
+            <h3 class="modal_header">Delete comment</h3>
+            <p class="modal_text">
+              Are you sure you want to delete this comment? This will remove the
+              comment and can’t be undone.
+            </p>
+            <div class="modal_buttons">
+              <button
+                class="modal_btn modal_cancel"
+                type="button"
+                @click="setShowModal(false)"
+              >
+                NO, CANCEL
+              </button>
+              <button
+                class="modal_btn modal_delete"
+                type="button"
+                @click="setShowModal(false), deleteComment(comment.id)"
+              >
+                YES, DELETE
+              </button>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+    <div class="reply_container">
+      <form
+        v-if="isReplying"
+        @submit.prevent="addAnswerHandler"
+        class="answer_form"
+      >
+        <textarea
+          v-model="replyText"
+          class="message_textarea"
+          name="message"
+          id="message"
+          cols="30"
+          rows="10"
+          placeholder="Add a reply…"
+          required
+        />
+        <img
+          class="avatar"
+          width="32"
+          height="32"
+          :src="require(`../photos/photosForProfile/${currentUser.pathToPhoto}`)"
+          alt="avatar"
+        />
+        <button class="send_message" type="submit">REPLY</button>
+      </form>
+      <div v-if="comment.answers && comment.answers.length !== 0">
+        <Message
+          :comment="answer"
+          v-for="answer in comment.answers"
+          :key="answer.id"
+        />
+      </div>
+    </div>
+  </template>
+  <script>
+  import moment from "moment";
+  import { v4 } from "uuid";
+  import ReferenceChecking from "./referenceChecking";
+  import { mapState, mapMutations } from "vuex";
+  import Delete from "../icons/delete";
+  import Edit from "../icons/edit";
+  import Message from "./message";
+  export default {
+    components: {
+      ReferenceChecking,
+      Delete,
+      Edit,
+      Message,
+    },
+    name: "Message",
+    props: {
+      comment: {
+        type: Object,
+        required: true,
+      },
+    },
+    data() {
+      return {
+        showModal: false,
+        isEditable: false,
+        isReplying: false,
+        textareaValue: "",
+        replyText: "",
+      };
+    },
+    methods: {
+      dateDifference(startDate) {
+        const startMoment = moment(startDate, "YYYY-MM-DD HH:mm");
+        const timeAgo = startMoment.fromNow(true);
+        return timeAgo;
```

Comment:
no need for intermediary variables
```suggestion
        return moment(startDate, "YYYY-MM-DD HH:mm").fromNow(true);
```